### PR TITLE
Bring osu! iOS to .NET 6

### PR DESCRIPTION
**Pull request by frenzibyte**
- [x] Depends on https://github.com/ppy/osu-framework/pull/5582

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -132,8 +132,8 @@ jobs:
         with:
           dotnet-version: "6.0.x"
 
-      # Contrary to seemingly any other msbuild, msbuild running on macOS/Mono
-      # cannot accept .sln(f) files as arguments.
-      # Build just the main game for now.
+      - name: Install .NET Workloads
+        run: dotnet workload install maui-ios
```
Comment (smoogipoo) [inline .github/workflows/ci.yml:136]: `maui-ios`? Just `ios` doesn't work? (Also in other files)

File: osu.iOS/osu.iOS.csproj [hunk 2]
```diff
@@ -1,124 +1,16 @@
...
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup Label="Project">
+    <TargetFramework>net6.0-ios</TargetFramework>
+    <SupportedOSPlatformVersion>13.4</SupportedOSPlatformVersion>
     <OutputType>Exe</OutputType>
-    <ProjectGuid>{3F082D0B-A964-43D7-BDF7-C256D76A50D0}</ProjectGuid>
-    <RootNamespace>osu.iOS</RootNamespace>
-    <AssemblyName>osu.iOS</AssemblyName>
-    <MtouchUseLlvm>false</MtouchUseLlvm> <!-- This currently causes random lockups during gameplay. https://github.com/mono/mono/issues/18973 -->
+    <UseMauiEssentials>true</UseMauiEssentials>
```
Comment (smoogipoo) [inline osu.iOS/osu.iOS.csproj:6]: Weird, required? Leads to the using in OsuGameIOS.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.iOS/osu.iOS.csproj (hunk 2 above)
Comment (smoogipoo) [reply osu.iOS/osu.iOS.csproj:6]: I guess this is probably required for battery info or something... Kinda awkward but okay.

**Comment by peppy** [comment]
Merged into https://github.com/ppy/osu/pull/21629

**Review by peppy — APPROVED** [review]

**Pull request closed without merging**